doc/rtd/reference/custom_modules/custom_configuration_module.rst

```diff
@@ -0,0 +1,18 @@
+.. _custom_configuration_module:
+
+Custom Configuration Module
+***************************
+
+Custom 3rd-party out-of-tree configuration modules can be added to cloud-init
+by:
+
+#. :ref:`Implement a config module<module_creation>` in a Python file with its
+   name starting with ``cc_``.
+
+#. Place the file where the rest of config modules are
+   On Ubuntu this path is typically:
+   `/usr/lib/python3/dist-packages/cloudinit/config/`.
+
+#. Extend the :ref:`base-configuration's <base_config_module_keys>`
+   ``cloud_init_modules``, ``cloud_config_modules`` or ``cloud_final_modules``
+   to let the config module run on one of those stages.
```

Comment:
I was about to leave a comment about "don't forget to pass a configuration for your custom module to your instance", but that made me realize something else:

I don't think we have a story around jsonschema loading 3rd party configuration modules, do we? That is, 3rd parties would have to manually modify our schema in order to add 3rd party validation. Think that we should document this shortcoming and/or file a bug for it?

Reply:
https://github.com/canonical/cloud-init/issues/5558

Reply:
I have added a warning explaining that and the issue.